activerecord/test/cases/adapters/mysql2/mysql2_adapter_test.rb

```diff
@@ -19,6 +19,22 @@ def test_exec_query_nothing_raises_with_no_result_queries
     end
   end
 
+  def test_database_exists_returns_false_if_database_does_not_exist
+    error = ->(_) { raise Mysql2::Error.new("Unknown database")  }
```

Comment:
https://github.com/rails/rails/pull/36025#discussion_r292595529

Reply:
I think this comment might be useful: https://github.com/rails/rails/pull/36025#discussion_r293160859

Reply:
worked like a charm!